extensions/notebook/src/dialog/remoteBookDialog.ts

```diff
@@ -0,0 +1,343 @@
+			value: loc.invalidTextPlaceholder
+		});
+	}
+
+	private get remoteLocationValue(): string {
+		return (<azdata.CategoryValue>this.remoteLocationDropdown.value).name;
+	}
+
+	public onRemoteLocationChanged(): void {
+		if (this.controller.getReleases() !== undefined && this.remoteLocationValue === loc.onGitHub) {
+			this.releaseDropdown.enabled = true;
+		} else {
+			this.releaseDropdown.enabled = false;
+		}
+	}
+
+	public async validate(): Promise<void> {
+		try {
+			let url = utils.getDropdownValue(this.githubRepoDropdown);
+			url = url.trim().toLowerCase();
+			if (this.remoteLocationValue === loc.onGitHub && url.length > 0) {
+				//get the first group to extract /owner/repo/releases format
+				let groupsRe = url.match(this.urlGithubRE);
+				if (groupsRe !== null && groupsRe.length > 0) {
+					url = loc.apiGitHub(groupsRe[0]);
+					let releases = await this.controller.fetchGithubReleases(new URL(url));
+					if (releases) {
+						this.releaseDropdown.enabled = true;
+						await this.fillReleasesDropdown();
+					}
+				} else {
+					throw new Error(loc.urlGithubError);
+				}
+			}
+		}
+		catch (error) {
+			this.dialog.message = {
+				text: (typeof error === 'string') ? error : error.message,
+				level: azdata.window.MessageLevel.Error
+			};
+		}
+	}
+
+	public async getAssets(): Promise<void> {
+		try {
+			if (this.remoteLocationValue === loc.onGitHub) {
+				let selected_release = this.controller.getReleases().filter(release =>
+					release.name === this.releaseDropdown.value);
+				let assets = await this.controller.fecthListAssets(selected_release[0]);
+				if (assets !== undefined && assets.length > 0) {
+					let bookValues: string[] = ['-'];
+					assets.forEach(asset => {
+						bookValues.push(asset.book);
+					});
+					bookValues = [...new Set(bookValues)]; //Remove duplicates
+					this.bookDropdown.values = bookValues;
+				}
+			}
+		}
+		catch (error) {
+			this.setFieldsToEmpty();
+			this.dialog.message = {
+				text: (typeof error === 'string') ? error : error.message,
+				level: azdata.window.MessageLevel.Error
+			};
+		}
+	}
+
+	public async download(): Promise<boolean> {
+		try {
+			if (this.remoteLocationValue === loc.onGitHub) {
+				let selected_asset = await this.getSelectedAsset();
+				if (selected_asset === undefined) {
+					throw (msgUndefinedAssetError);
+				}
+				await this.controller.setRemoteBook(selected_asset.url, this.remoteLocationValue, selected_asset);
+			} else {
+				let url = utils.getDropdownValue(this.githubRepoDropdown);
+				let newUrl = new URL(url);
+				await this.controller.setRemoteBook(newUrl, this.remoteLocationValue);
+			}
+			return true;
+		}
+		catch (error) {
+			this.dialog.message = {
+				text: (typeof error === 'string') ? error : error.message,
+				level: azdata.window.MessageLevel.Error
+			};
+			return false;
+		}
+	}
+
+	public async fillReleasesDropdown(): Promise<void> {
+		let versions: string[] = ['-'];
```

Comment:
`this.releaseDropdown.values = ['-'].concat(this.controller().getReleases().map(release => release.name));`